### PR TITLE
Added Configurable Date Format For App

### PR DESCRIPTION
**Pull request by aurangzaibumer**
added date format configurable and applied on sync time in navigation, setting screen and insight screen

**IMPORTANT: Where possible all PRs must be linked to a Github issue**

Fixes : https://github.com/opensrp/fhircore/issues/3472

Screenshot (Where new date format has been applied)

image
image
image

**Engineer Checklist**
- [ ] I have written **Unit tests** for any new feature(s) and edge cases for bug fixes
- [ ] I have added any strings visible on UI components to the `strings.xml` file
- [ ] I have updated the  [CHANGELOG.md](./CHANGELOG.md) file for any notable changes to the codebase
- [ ] I have run `./gradlew spotlessApply` and `./gradlew spotlessCheck` to check my code follows the project's style guide
- [ ] I have built and run the FHIRCore app to verify my change fixes the issue and/or does not break the app 
- [ ] I have checked that this PR does NOT introduce **breaking changes** that require an update to **_Content_** and/or **_Configs_**? _If it does add a sample here or a link to exactly what changes need to be made to the content._

**Code Reviewer Checklist**
- [ ] I have verified **Unit tests** have been written for any new feature(s) and edge cases
- [ ] I have verified any strings visible on UI components are in the `strings.xml` file
- [ ] I have verifed the [CHANGELOG.md](./CHANGELOG.md) file has any notable changes to the codebase
- [ ] I have verified the solution has been implemented in a configurable and generic way for reuseable components
- [ ] I have built and run the FHIRCore app to verify the change fixes the issue and/or does not break the app

### CODE REVIEW

**Comment by dubdabasoduba** [comment]
@marklosh @Lentumunai-Mark  This seems to work well

date-app-version
date-insights-page
date-sync-bar

File: android/quest/src/test/java/org/smartregister/fhircore/quest/ui/main/AppMainViewModelTest.kt [hunk 1]
```diff
@@ -314,6 +317,33 @@ class AppMainViewModelTest : RobolectricTest() {
       }
     }
 
+  @Test
+  fun testGetSyncTimeWithMillisTimestamp() {
+    // Mocking a timestamp in milliseconds
+    val mockTimestamp = OffsetDateTime.now().toInstant().toEpochMilli().toString()
+    every { appMainViewModel.retrieveLastSyncTimestamp() } returns mockTimestamp
+    every { appMainViewModel.applicationConfiguration.dateFormat } returns "yyyy-MM-dd HH:mm:ss"
```
Comment (ellykits) [inline android/quest/src/test/java/org/smartregister/fhircore/quest/ui/main/AppMainViewModelTest.kt:325]: It is not necessary to mock this. Save and use the timestamp instead.

File: android/quest/src/test/java/org/smartregister/fhircore/quest/ui/main/AppMainViewModelTest.kt [hunk 2]
```diff
@@ -314,6 +317,33 @@ class AppMainViewModelTest : RobolectricTest() {
...
+    val expectedFormattedDate =
+      formatDate(mockTimestamp.toLong(), appMainViewModel.applicationConfiguration.dateFormat)
+    Assert.assertEquals(expectedFormattedDate, syncTime)
+  }
+
+  @Test
+  fun testGetSyncTimeWithFormattedDate() {
+    val mockFormattedDate = "2023-10-10 10:10:10"
+    every { appMainViewModel.retrieveLastSyncTimestamp() } returns mockFormattedDate
+    every { appMainViewModel.applicationConfiguration.dateFormat } returns "yyyy-MM-dd"
```
Comment (ellykits) [inline android/quest/src/test/java/org/smartregister/fhircore/quest/ui/main/AppMainViewModelTest.kt:336]: It is not necessary to mock this. Save and use the timestamp instead.

**Review by ellykits — CHANGES_REQUESTED** [review]

**Comment by aurangzaibumer** [comment]
Thank you for merging this PR.

**Pull request merged**